s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py

```diff
@@ -69,37 +78,80 @@ def uttest(self, index: str):
-    parser.add_argument("--key", help='Provide a key to be used in --get')
+    parser.add_argument("--set", help='Sets config value for this given index using a key, pass --key and --setval')
+    parser.add_argument("--persistent", help='Optional, pass <Yes> if needed, to be used with --set, this updates the backend along with --set')
+    parser.add_argument("--key", help='Provide a key to be used in --get and --set')
     parser.add_argument("--getkey", help='Fetch value of the given key from default index, pass --path to config')
-    parser.add_argument("--path", help='example: JSON:///etc/cortx/myconf.json')
-    parser.add_argument("--uttest", help='An unit test for the load, get and get_data APIs')
-
-    args = parser.parse_args()
+    parser.add_argument("--setkey", help='Sets config value for the given key from default index, pass --path to config and --setval')
+    parser.add_argument("--setval", help='String value to be set using --set or --setkey')
+    parser.add_argument("--path", help='Path to config file, supported formats are: json, toml, yaml, ini, pillar (salt). Usage: <format>://<file_path>, e.g. JSON:///etc/cortx/myconf.json')
+    parser.add_argument("--uttest", help='An unit test for the load, get, get_data and set APIs')
 
     if len(sys.argv) < 2:
       print("Incorrect args, use -h to review usage")
       exit(-1)
 
+    args = parser.parse_args()
+
     if args.path:
+      if args.load == None and args.getkey == None and args.setkey == None:
+        print("--path is needed by only --load, --getkey and --setkey")
+        exit(-1)
       if os.path.isfile(urlparse(args.path).path) != True:
         print("config file: {} does not exist".format(args.path))
         exit(-1)
       else:
-        try:
-          with open(urlparse(args.path).path) as f:
-            json.load(f)
-        except ValueError as e:
-          print("config file: {} must use valid JSON format: {}".format(urlparse(args.path).path, e))
+        store_type = urlparse(args.path).scheme
+        is_valid_type = False
+        valid_types = ''
+        storage = inspect.getmembers(kv_store_collection, inspect.isclass)
+        for name, cls in storage:
+          if hasattr(cls, 'name') and name != "KvStore":
+            valid_types += cls.name + ' '
+            if store_type == cls.name:
+              is_valid_type = True
+              break
+
+        if is_valid_type == False:
+          print("Invalid storage type {} in config file: {}, accepted types are {}".format(store_type, args.path, valid_types))
           exit(-1)
 
+        if store_type == 'json':
+          try:
+            with open(urlparse(args.path).path) as f:
+              json.load(f)
+          except ValueError as e:
+            print("config file: {} must use valid JSON format: {}".format(urlparse(args.path).path, e))
+            exit(-1)
+        """TODO: Implement rest of the type's content validators here"""
```

Comment:
add else part to log message that the store_type of the url is not supported

Reply:
If we do that, it will disable the current handling of other formats, such as yaml. Will it be ok?

Reply:
This is list of supported formats by conf_store are: dict ini json jsonmessage salt text toml yaml